### ISSUE TITLE
Lock Renewal behavior is not as expected

### ISSUE THREAD

**Issue by ramya0820** [Azure/azure-service-bus-node#231]
According to the documentation, message is supposed to expire after 5 minutes (300 seconds) by default.
It does not and test-case at #226 captures this scenario.

**Comment by AlexGhiondea** [comment]
Is it possible that the undefined value is treated as the default value and you end up with auto renewal of the lock?

**Comment by ramya0820** [comment]
> Is it possible that the undefined value is treated as the default value and you end up with auto renewal of the lock?

Yes, the default value is assumed was expected to be 300 seconds. But the lock doesn't expire after 300 seconds. May need to run more tests to see when does the lock expire, if at all. (It may expire after 400 seconds, or some other multiple of 20 seconds, as was observed in other test-cases)

**Comment by ramya-rao-a** [comment]
The lock may be renewed just before the 300 secs was up, therefore technically keeping the lock alive till about 330 seconds.

Once we have access to the lock duration by which the `renewLock` extends the lock time, we can do better estimations of this i.e say `x` is the lock duration. We avoid renewing the lock if current time is more than configured value minus x seconds.

Moving this to Preview 2, to revisit once we have the management plane via which we can get the lock duration
